tests/e2e/pageobjects/dashboard/CreateWorkspace.ts

```diff
@@ -14,53 +14,95 @@ import { DriverHelper } from '../../utils/DriverHelper';
 import { By, Key } from 'selenium-webdriver';
 import { Logger } from '../../utils/Logger';
 import { TimeoutConstants } from '../../TimeoutConstants';
+import { EditorType, TestConstants } from '../../TestConstants';
 
 @injectable()
 export class CreateWorkspace {
     static readonly FACTORY_URL_LOCATOR: By = By.xpath(`//input[@id="git-repo-url"]`);
 
     constructor(@inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) { }
 
-    async waitTitleContains(expectedText: string, timeout: number = TimeoutConstants.TS_COMMON_DASHBOARD_WAIT_TIMEOUT) {
+    async waitTitleContains(expectedText: string, timeout: number = TimeoutConstants.TS_COMMON_DASHBOARD_WAIT_TIMEOUT): Promise<void> {
         Logger.debug(`CreateWorkspace.waitTitleContains text: "${expectedText}"`);
 
         const pageTitleLocator: By = By.xpath(`//h1[contains(text(), '${expectedText}')]`);
 
         await this.driverHelper.waitVisibility(pageTitleLocator, timeout);
     }
 
-    async waitPage(timeout: number = TimeoutConstants.TS_SELENIUM_LOAD_PAGE_TIMEOUT) {
+    async waitPage(timeout: number = TimeoutConstants.TS_SELENIUM_LOAD_PAGE_TIMEOUT): Promise<void> {
         Logger.debug('CreateWorkspace.waitPage');
 
         await this.waitTitleContains('Create Workspace', timeout);
     }
 
-    async waitSample(sampleName: string, timeout: number = TimeoutConstants.TS_COMMON_DASHBOARD_WAIT_TIMEOUT) {
+    async waitSample(sampleName: string, timeout: number = TimeoutConstants.TS_COMMON_DASHBOARD_WAIT_TIMEOUT): Promise<void> {
         Logger.debug(`CreateWorkspace.waitSample sampleName: "${sampleName}"`);
 
         const sampleLocator: By = this.getSampleLocator(sampleName);
 
         await this.driverHelper.waitVisibility(sampleLocator, timeout);
     }
 
-    async clickOnSample(sampleName: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT) {
+    async clickOnSample(sampleName: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT): Promise<void> {
```

Comment:
Same here, I would rename it to `clickOnSampleUsingDefaultEditor` or something similar and replace it's use with `clickOnSampleForSpecificEditor` in most of the logic.

We can use this for a specific "default editor test" later

Reply:
name changed